Sudoku/buildFunctions.js

```diff
@@ -0,0 +1,74 @@
+const {SQUARES, BOARD_LENGTH, NUMBERS_PER_SQUARE} = require('./constants')
+
+const buildSquareContainer = (arrayOrObject, board) => {
+  const bigSquareContainer = arrayOrObject === 'array' ? [] : {}
+  for(let bigSquare in SQUARES) {
+    let iterI = 0
+    const smallSquareContainer = arrayOrObject === 'array' ? [] : {}
+    for(let smallSquare in SQUARES[ bigSquare ]) {
+      const location = SQUARES[ bigSquare ][ smallSquare ]
+      const smallIndex = arrayOrObject === 'array' ? iterI++ : smallSquare
+      smallSquareContainer[ smallIndex ] = board[ location ]
+    }
+    if(arrayOrObject === 'array'){
+      bigSquareContainer.push( smallSquareContainer )
+    } else {
+      bigSquareContainer[ bigSquare ] = smallSquareContainer
+    }
+  }
+
+  return bigSquareContainer
+}
+
+const buildRowContainer = board => {
+  const arr = []
+  for(let itJ = 0; itJ < 9; itJ++) {
+    arr.push( board.substring(itJ*9, (itJ*9)+9).split('') )
+  }
```

Comment:
Yes, indeed I was. And thus have refactored since, only after noticing I was doing just that.